src/lib/raft/raft.c

```diff
@@ -1006,8 +1109,8 @@ raft_cfg_election_timeout(struct raft *raft, double timeout)
 		return;
 
 	raft->election_timeout = timeout;
-	if (raft->vote == 0 || raft->leader != 0 || !raft->is_candidate ||
-	    raft->is_write_in_progress)
+	if (raft->vote == 0 || raft_is_leader_seen(raft) ||
+	    !raft->is_candidate || raft->is_write_in_progress)
```

Comment:
This change doesn't seem to be tested. I reverted it and the unit tests passed.

Reply:
This was actually a malicious change. So, thanks for noticing!
I've covered it with a test and reverted this line:
```diff
diff --git a/src/lib/raft/raft.c b/src/lib/raft/raft.c
index 2177cc2bc..a7108e144 100644
--- a/src/lib/raft/raft.c
+++ b/src/lib/raft/raft.c
@@ -1151,8 +1151,8 @@ raft_cfg_election_timeout(struct raft *raft, double timeout)
                return;
 
        raft->election_timeout = timeout;
-       if (raft->vote == 0 || raft_is_leader_seen(raft) ||
-           !raft->is_candidate || raft->is_write_in_progress)
+       if (raft->vote == 0 || raft->leader != 0 || !raft->is_candidate ||
+           raft->is_write_in_progress)
                return;
 
        assert(raft_ev_timer_is_active(&raft->timer));

```